agent/qrexec-agent-data.c

```diff
@@ -598,30 +636,38 @@ int handle_new_process_common(int type, int connect_domain, int connect_port,
         case MSG_JUST_EXEC:
             send_exit_code(data_vchan, handle_just_exec(cmdline));
             break;
-        case MSG_EXEC_CMDLINE:
-            do_fork_exec(cmdline, &pid, &stdin_fd, &stdout_fd, &stderr_fd);
+        case MSG_EXEC_CMDLINE: {
+            struct buffer stdin_buf;
+            if (execute_qubes_rpc_command(cmdline, &pid, &stdin_fd, &stdout_fd, &stderr_fd, !qrexec_is_fork_server, &stdin_buf) < 0) {
+                fputs("failed to spawn process\n", stderr);
+                return 126;
```

Comment:
This changes the semantic in cases like "File not found" (no exit code send) or did I miss something?